provider/provider.go

```diff
@@ -31,6 +31,7 @@ func GetResourceType(name string) (tftypes.Type, error) {
 func GetProviderResourceSchema() map[string]*tfprotov5.Schema {
 	waitForType := tftypes.Object{
 		AttributeTypes: map[string]tftypes.Type{
+			"timeout": tftypes.String,
```

Comment:
We need to make this optional, but we're currently blocked because of this: https://github.com/hashicorp/terraform-plugin-go/issues/83#issuecomment-856638794